### DISCUSSION TITLE
Is there a way to enable the old scrubbing behavior?

### DISCUSSION THREAD

**Discussion by Thane5** [ggarra13/mrv2#180] [General]
I noticed that since v0.8.2, scrubbing with the mouse on the image doesnt linearly scrub anymore, but instead starts to play the video either forwards or backwards. 
I know there is a workaround, where if i mute the audio, the old behavior is back again, but i'd much prefer to be able to have it do that by default, and without disabling the audio.

So, in case i missed it, is there a setting in the preferences for that?

**Comment by ggarra13** [comment]
No.  Just to clarify.  The old v0.8.2 behavior would not scrub with audio by default.  Only if you had first started playing the clip, then it would scrub it with audio (but always playing forwards).  Is that behavior that what you want back?

**Comment by Thane5** [comment]
I don't care about the audio so much, i just like directly clicking ont he video to scrub the video back and forth, because this is much faster and more precise than using the timeline for that (This recording is from v0.8.1)

https://github.com/ggarra13/mrv2/assets/13683581/9a0d66f3-739a-4105-b7b6-963d405010b9

**Comment by ggarra13** [comment]
Okay, find in sourceforge, the beta folder a new version on Windows 8.1+, Ubuntu 20.04.3 LTS+ and soon on macOS 11+ that should contain a Preferences->Playback called Auto Scrub Playback that when turned OFF should made it behave like the old mrv2.

**Comment by AltaArts** [comment]
Sorry about reviving an old thread, but was wondering if this is still the plan?  I too would love the ability to jog/scrub with the mouse but not have it keep playing when the mouse stops.  Also, I would map this to my external control panel's jog wheel - instead of mapping the keyboard shortcuts.

I am using v 1.20 (mrv2 64 bits - v1.2.0 - Built Jun 25 2024 23:20:10), and also tried it in 1.19.  I do not see a Preference "Auto Scrub Playback".  But maybe I am doing something incorrect?

Thanks much,

Joshua Breckeen.

**Comment by Thane5** [comment]
well thats weird, i just installed mvr2 on another device and cant find that setting either.

**Comment by AltaArts** [comment]
Lol, yeah I have been there.  One time it was just because I had commented out a section in testing, and forgot to re-enable - doh.

Thanks much.

**Comment by ggarra13** [comment]
Sorry guys.  The setting is now named Preferences->Playback->Scrub with Audio.

**Comment by AltaArts** [comment]
Ah, yes it is!  Thanks so much, it works well.

J.

**Comment by Thane5** [comment]
May i ask why you chose to make this new scrubbing behavior the default? I don't really see it as vastly more useful than the old one. Personally i always loved how that was basically a replacement to scrubbing with the timeline, but much faster and more intuitive.

**Comment by ggarra13** [comment]
Someone asked me to scrub with audio, so that meant that I had to play the movie while scrubbing as that's a current limitation of Darby's tlRender library.  
What I forgot to do was check if the movie did *NOT* have audio, so that the old scrubbing behavior was used.  I have now addressed that for v1.2.1, which I will release shortly, as v1.2.0 also broke all external codec libraries ( Sigh :( ).  Hopefully, you will like that behavior.

**Comment by Thane5** [comment]
In that case, is it intended to act as a "play forward/backwards" button? Because when i move the mouse in one direction and then stop, it will simply keep playing until i stop holding the mouse down?

**Reply by ggarra13** [reply]
> In that case, is it intended to act as a "play forward/backwards" button? Because when i move the mouse in one direction and then stop, it will simply keep playing until i stop holding the mouse down?

No.  That would be a bug.  I think it was in all versions that scrubed with audio.  However, if you know of a previous version that did not do that and that scrubbed with audio, please let me know so I can check what I broke.

**Comment by ggarra13** [comment]
Actually, I just checked with the bbb_sunflower* movie, and I cannot reproduce the behavior you describe (it keeps playing forward only if you were playing before, but if you are stopped it does not keep playing).  

Please open an Issue with steps to reproduce it, and hopefully, a movie where you see the problem.  I will look into it.
